src/core/Akka.Streams/Stage/GraphStage.cs

```diff
@@ -1669,7 +1689,14 @@ protected void PassAlong<TOut, TIn>(Inlet<TIn> from, Outlet<TOut> to, bool doFin
         /// <param name="handler">TBD</param>
         /// <returns>TBD</returns>
         protected Action<T> GetAsyncCallback<T>(Action<T> handler)
-            => @event => Interpreter.OnAsyncInput(this, @event, x => handler((T)x));
+            => @event =>
+            {
+                if(_interpreter == null)
+                    _callbacksWaitingForInterpreter.Enqueue(
+                        new ActorGraphInterpreter.AsyncInput(null, null, @event, NoPromise, x => handler((T)x)));
+                else
+                    Interpreter.OnAsyncInput(this, @event, NoPromise, x => handler((T)x));
+            };
```

Comment:
Example of how the invocation is stored using the existing `AsyncInput` data structure to be defered for later. The actual execution will be performed when `Interpreter` property is assigned.